apps/main/src/dex/components/PagePool/Page.tsx

```diff
@@ -3,57 +3,59 @@ import { useRouter } from 'next/navigation'
 import { useEffect, useMemo } from 'react'
 import Transfer from '@/dex/components/PagePool/index'
 import { ROUTE } from '@/dex/constants'
-import usePageOnMount from '@/dex/hooks/usePageOnMount'
 import useStore from '@/dex/store/useStore'
-import type { PoolUrlParams } from '@/dex/types/main.types'
-import { getPath } from '@/dex/utils/utilsRouter'
-import { useApiStore } from '@ui-kit/shared/useApiStore'
+import type { CurveApi, PoolUrlParams } from '@/dex/types/main.types'
+import { getPath, useChainId } from '@/dex/utils/utilsRouter'
+import { isLoading, useConnection } from '@ui-kit/features/connect-wallet'
 
-const Page = (params: PoolUrlParams) => {
+export const PagePool = (props: PoolUrlParams) => {
   const { push } = useRouter()
-  const { pageLoaded, routerParams, curve } = usePageOnMount()
-  const { rChainId, rPoolId, rFormType } = routerParams
+  const { lib: curve = null, connectState } = useConnection<CurveApi>()
+  const {
+    pool: rPoolId,
+    formType: [rFormType],
```

Comment:
Same potential destructure here. Keeping an eye open

Reply:
As I said it can be used inside the [`[...formType]` folder](https://github.com/curvefi/curve-frontend/blob/e5fdad7d72349de53300c6d89589182ea342e0d0/apps/main/src/app/dex/%5Bnetwork%5D/pools/%5Bpool%5D/%5B%5B...formType%5D%5D/page.tsx#L8)